### PR TITLE
Enable TLS 1.3 support

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by gearama — APPROVED** [review]

File: sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp [hunk 1]
```diff
@@ -723,8 +723,8 @@ Azure::Core::_internal::UniqueHandle<HINTERNET> WinHttpTransport::CreateSessionH
       sizeof(tls_false_start));
 #endif
 
-  // Enforce TLS version 1.2
-  auto tlsOption = WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2;
+  // Enforce TLS version 1.2 or 1.3
+  auto tlsOption = WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2 | WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_3;
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp:727]: Update error message below: `Error while enforcing TLS 1.2 or 1.3 ...`

**Review by ahsonkhan — APPROVED** [review]
Other than updating the error message, looks good.

**Comment by Jinming-Hu** [comment]
What's the storage for libcurl? Is TLS1.3 enabled by default in libcurl if the version is high enough?

**Comment by LarryOsterman** [comment]
> What's the story for libcurl? Is TLS1.3 enabled by default in libcurl if the version is high enough?

CURL supports TLS 1.2 and higher (which means TLS 1.3 on CURL version 7.52 and higher).

If the customer is using CURL versions lower than 7.52 then they don’t have TLS 1.3 support.

We don't control which versions of CURL the customer chooses to use unfortunately.

**Pull request merged**